rows.go

```diff
@@ -80,24 +106,63 @@ func (rs *rowSets) empty() bool {
 	return true
 }
 
+// Column is a mocked column Metadate for rows.ColumnTypes()
+type Column struct {
+	name, dbTyp              string
+	nullable                 bool
+	length, precision, scale int64
+	scanType                 reflect.Type
+}
+
 // Rows is a mocked collection of rows to
 // return for Query result
 type Rows struct {
 	converter driver.ValueConverter
 	cols      []string
+	def       []*Column
 	rows      [][]driver.Value
 	pos       int
 	nextErr   map[int]error
 	closeErr  error
 }
 
+// New Column allows to create a Column Metadata definition
+func NewColumn(name, dbTyp string, exampleValue interface{}, nullable bool, length, precision, scale int64) *Column {
+	return &Column{name, dbTyp, nullable, length, precision, scale, reflect.TypeOf(exampleValue)}
+}
+
+func NewColumnSimple(name string) *Column {
+	return &Column{name: name}
+}
+
 // NewRows allows Rows to be created from a
 // sql driver.Value slice or from the CSV string and
 // to be used as sql driver.Rows.
 // Use Sqlmock.NewRows instead if using a custom converter
 func NewRows(columns []string) *Rows {
+	definition := make([]*Column, len(columns))
+	for i, column := range columns {
+		definition[i] = NewColumnSimple(column)
+	}
+
 	return &Rows{
 		cols:      columns,
+		def:       definition,
+		nextErr:   make(map[int]error),
+		converter: driver.DefaultParameterConverter,
+	}
+}
+
+// NewRowsWithColumnDefiniton see PR-152
+func NewRowsWithColumnDefiniton(columns ...*Column) *Rows {
```

Comment:
sqlmock interface can also [initialize Rows](https://github.com/DATA-DOG/go-sqlmock/blob/master/sqlmock.go#L76) with an user defined column converter. The same method should be there too.